draft-ietf-rtcweb-jsep.xml

```diff
@@ -556,6 +556,82 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
 
       </section>
 
+      <section anchor="sec.imageattr" title="Video Size Negotiation">
+        <t>Video size negotiation is the process through which a receiver can
+          use the "a=imageattr" SDP attribute <xref target="RFC6236"/> to indicate
+          what video frame sizes it is capable of receiving. A receiver may have
+          hard limits on what its video decoder can process, or it may wish to
+          constrain what it receives due to application preferences, e.g. a
+          specific size for the window in which the video will be displayed.
+        </t>
+
+        <section title="Creating an imageattr Attribute">
+          <t>In order to determine the limits on what video resolution a
+            receiver wants to receive, it will intersect its decoder hard
+            limits with any mandatory constraints that have been applied to
+            the associated MediaStreamTrack. For the answerer, these mandatory
+            constraints can be applied to the remote MediaStreamTracks that
+            are created by a setRemoteDescription call, and will affect the
+            output of the ensuing createAnswer call. For the offerer, since it
+            does not know about any remote MediaStreamTracks until it receives
+            theanswer, this implies that it can only set these constraints after
+            the initial offer-answer exchange is complete, and as a result, a
+            new offer-answer is needed to communicate them.</t>
```

Comment:
OK - I think I'm starting to get one this is getting at. Let's separate two cases here. Case 1 is A is mobile phone that the actually browser knows that due to hardware limitations it can not receive more than 360p, and case 2 where it's a desktop browser that can receive big stuff but the JS app wants to limit this particular video stream to 360p.

In case 1, the limit that A can only receive 360p would have been in the initial Offer. 

In case 2, what you are saying is that as the JS API layer, we have no way to set this limit until after an answer or pr-answer is received by A? It seems we need to fix this API layer not so much for this but more for simulcast. 

Agree of course that any time A wants to change it's constraints after sending the Offer, it's going to need to do new O/A exchange.